src/libsyntax/errors/emitter.rs

```diff
@@ -361,16 +361,28 @@ fn print_diagnostic(dst: &mut Destination,
                     msg: &str,
                     code: Option<&str>)
                     -> io::Result<()> {
-    if !topic.is_empty() {
-        dst.start_attr(term::Attr::ForegroundColor(lvl.color()))?;
-        write!(dst, "{}: ", topic)?;
-        dst.reset_attrs()?;
-    }
 
-    dst.start_attr(term::Attr::Bold)?;
-    write!(dst, "{}: ", lvl.to_string())?;
-    write!(dst, "{}", msg)?;
-    dst.reset_attrs()?;
+
+    match lvl {
+        Level::Error => {
```

Comment:
hmm, one concern I have is that if the "top-level" error is a warning, this won't get bolded. I put in some logic to emit newlines only before the *start* of an error -- perhaps we should modify that to push down a boolean flag, and use that flag as the criteria here?